pkg/stanza/fileconsumer/config.go

```diff
@@ -101,7 +103,11 @@ func (c Config) Build(logger *zap.SugaredLogger, emit EmitFunc) (*Manager, error
 	}
 
 	// Ensure that splitter is buildable
-	factory := newMultilineSplitterFactory(c.Splitter.EncodingConfig, c.Splitter.Flusher, c.Splitter.Multiline)
+	var factory splitterFactory
+	factory = newDefaultSplitterFactory(c.EncodingConfig, c.Flusher)
+	for _, opt := range opts {
+		factory = opt(c.EncodingConfig, c.Flusher)
```

Comment:
We have to pass flusher in to create the factory, and again to modify the factory. Is there a way to avoid that?

Reply:
let me see